-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-48764/45837

ERROR: Build errors found during clang-tidy run.
```
src/SimG4Core/Application/src/Phase2SteppingAction.cc:202:43: error: use of undeclared identifier 'doFineCalo_' [clang-diagnostic-error]
  202 |       if (!trkinfo->crossedBoundary() && !doFineCalo_) {
      |                                           ^
Suppressed 697 warnings (697 in non-user code).
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/SimG4Core/Application/interface/Phase2SteppingAction.h
+++ b/SimG4Core/Application/interface/Phase2SteppingAction.h
@@ -57,6 +57,7 @@ private:
   bool killBeamPipe{false};
   bool hasWatcher;
   bool dd4hep_;
+  bool doFineCalo_;
 
   std::vector<double> maxTrackTimes, ekinMins;
   std::vector<std::string> maxTimeNames, ekinNames, ekinParticles;

--- a/SimG4Core/Application/python/g4SimHits_cfi.py
+++ b/SimG4Core/Application/python/g4SimHits_cfi.py
@@ -347,6 +347,7 @@ g4SimHits = cms.EDProducer("OscarMTProducer",
         EndPrintTrackID = cms.int32(0)
     ),
     SteppingAction = cms.PSet(
+        common_MCtruth,
         common_maximum_time,
         CMStoZDCtransport       = cms.bool(False),
         MaxNumberOfSteps        = cms.int32(20000),

--- a/SimG4Core/Application/src/Phase2SteppingAction.cc
+++ b/SimG4Core/Application/src/Phase2SteppingAction.cc
@@ -38,6 +38,7 @@ Phase2SteppingAction::Phase2SteppingAction(const CMSSteppingVerbose* sv,
   caloName_ = (G4String)(p.getParameter<std::string>("CaloName"));
   btlName_ = (G4String)(p.getParameter<std::string>("BTLName"));
   cms2ZDCName_ = p.getParameter<std::string>("CMS2ZDCName");
+  doFineCalo_ = (p.getParameter<bool>("DoFineCalo"));
 
   edm::LogVerbatim("SimG4CoreApplication")
       << "Phase2SteppingAction:: KillBeamPipe = " << killBeamPipe

